E2E: Product Query: Variations tests

dinhtungdu opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->

<!-- Reference any related issues or PRs here -->

This PR adds E2E tests for Product Query variations including Products On Sale.

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

Run test locally:
```
node bin/wp-env-with-gutenberg.js
GUTENBERG_EDITOR_CONTEXT=gutenberg npm run test:e2e tests/e2e/specs/backend/product-query/variations.test.ts
```

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

dinhtungdu committed:
add e2e tests for on sale variation of product query

dinhtungdu committed:
refactor common utils and selector

dinhtungdu committed:
delete out of date tests

dinhtungdu committed:
change test match config to include only test files with suffix

dinhtungdu commented on tests/e2e/config/jest.config.js:
```diff
@@ -32,4 +32,5 @@ module.exports = {
 	],
 	testPathIgnorePatterns: [ '<rootDir>/tests/e2e/specs/performance' ],
 	transformIgnorePatterns: [ 'node_modules/(?!(woocommerce)/)' ],
+	testMatch: [ '**/?(*.)+(spec|test).[jt]s?(x)' ],
```
Currently, we're using the config from `@wordpress/scripts`, which is:

```
	testMatch: [ '**/specs/**/*.[jt]s?(x)', '**/?(*.)spec.[jt]s?(x)' ],
```

I changed it to use one of Jest's [default match expressions](https://jestjs.io/docs/configuration#testmatch-arraystring). By doing so, we only match the test files with valid suffixes.

dinhtungdu committed:
Merge branch 'trunk' into tests/product-query-variations

dinhtungdu committed:
finish renaming product filters to advanced filters

dinhtungdu committed:
Merge branch 'trunk' into tests/product-query-variations

sunyatasattva reviewed (commented):
Should we put this PR `on hold`? It looks like we might remove the variations next week…

dinhtungdu commented:
> Should we put this PR `on hold`? It looks like we might remove the variations next week…

@sunyatasattva Sure, I just added `oh hold` label to this PR.

Can you take a look at other parts of this PR, beside the variation tests, I refactored the structure of other tests to make the code more reusable.

sunyatasattva commented:
@dinhtungdu checked the other improvements and I like them very much! Test code is becoming cleaner and cleaner, thank you!

dinhtungdu commented:
> @dinhtungdu checked the other improvements and I like them very much! Test code is becoming cleaner and cleaner, thank you!

Yay! I will move that part to another/new PR so we can merge them sooner.

dinhtungdu committed:
Merge branch 'trunk' into tests/product-query-variations

sunyatasattva commented:
Since you have moved the useful changes from this, and we have decided (p1670535184371009-slack-C02FL3X7KR6) to remove the variations, we can get rid of the rest of the code.
